guides/specification.md

```diff
@@ -0,0 +1,555 @@
+
+```yaml
+expectations:
+  - name: <expectation_name>
+    expect: <expectation_expression>
+
+  - name: <another_expectation_name>
+    expect: <another_expectation_expression>
+
+  - name: <yet_another_expectation_name>
+    expect_same: <yet_another_expectation_expression>
+```
+
+Extra considerations:
+- there can be many expectations for a single Check
+- an expectation can be one of two types [`expect`](#expect) or [`expect_same`](#expect-same)
+- a Check passes when all the expectations are satisfied 
+
+Example
+```yaml
+expectations:
+  - name: token_timeout
+    expect: facts.corosync_token_timeout == values.expected_token_timeout
+
+  - name: awesome_expectation
+    expect: facts.awesome_fact == values.awesome_expected_value
+```
+
+In the previous example a Checks passes (is successful) if all expectations are met, meaning that
+```
+facts.corosync_token_timeout == values.expected_token_timeout
+AND
+facts.awesome_fact == values.awesome_expected_value
+```
+### Expect
+
+This type of expectation is satisfied when, after facts gathering, the expression is `true` for all the targets involved in the current execution.
+
+> Execution Scenario:
+> 
+> - 2 targets [`A`, `B`]
+> - selected Checks [`corosync_check`]
+> - some environment (context)
+> 
+> ```yaml
+> expectations:
+>   - name: token_timeout
+>     expect: facts.corosync_token_timeout == values.expected_token_timeout
+> ```
+
+Considering the previous scenario what happens is that:
+- the fact `corosync_token_timeout` is gathered on all targets (`A` and `B` in this case)
+- the expectation expression gets executed against the `corosync_token_timeout` fact gathered on every targets.
+  - `targetA.corosync_token_timeout == values.expected_token_timeout`
+  - `targetB.corosync_token_timeout == values.expected_token_timeout`
+- every evaluation has to be `true`
+
+### Expect Same
+
+This type of expectation is satisfied when, after facts gathering, the expression's return value is the same for all the targets involved in the current execution.
+
+> Execution Scenario:
+> 
+> - 2 targets [`A`, `B`]
+> - selected Checks [`some_check`]
+> - some environment (context)
+> 
+> ```yaml
+> expectations:
+>   - name: awesome_expectation
+>     expect_same: facts.awesome_fact
+> ```
+
+Considering the previous scenario what happens is that:
+- the fact `awesome_fact` is gathered on all targets (`A` and `B` in this case)
+- the expectation expression gets executed for every target involved.
+  - `targetA.facts.awesome_fact`
+  - `targetB.facts.awesome_fact`
+- the expressions results has to be the same for every target
+  - `targetA.facts.awesome_fact == targetB.facts.awesome_fact`
+
+## Expression Language
+
+Different parts of the Check declaration are places where a computation is needed.
+
+> Provide a contextual [Description](#description)/[Remediation](#remediation) text, based on the execution context
+> 
+> interpolation of `{{ <expression> }}`
+```yaml
+description: |
```

Comment:
Sorry for being insistent, but I wouldn't put interpolations in descriptions as example